src/lib.rs

```diff
@@ -146,16 +146,67 @@ fn get_num_cpus() -> usize {
     cpus as usize
 }
 
-#[cfg(
-    any(
-        target_os = "linux",
-        target_os = "nacl",
-        target_os = "macos",
-        target_os = "ios",
-        target_os = "android",
-        target_os = "solaris",
-        target_os = "fuchsia",
-    )
+#[cfg(feature = "nightly")]
+#[feature(asm, naked_functions)]
```

Comment:
This function exists in the standard library: https://doc.rust-lang.org/std/primitive.u64.html#method.count_ones